src/game/Game.kt

```diff
@@ -2,31 +2,35 @@ package game
 
 import action.Action
 import agent.Player
+import exception.ExceptionMessages.Companion.HIT_WALL_EXCEPTION_MESSAGE
+import exception.ExceptionMessages.Companion.WALL_FEEDBACK
+import exception.InvalidMoveException
 import map.ExitNode
 import map.WorldMap
-import java.lang.UnsupportedOperationException
 
 class Game(private val player: Player, private val worldMap: WorldMap) {
 
     var timeCounter: Int = 0
 
     fun run() {
         while(!Thread.interrupted()) {
-            val nextMove = player.chooseNextMove()
-            when (nextMove) {
-                Action.MOVE_NORTH ->
-                    worldMap.moveObject(player.actor, 0, -1)
-                else -> {
-                    throw UnsupportedOperationException()
```

Comment:
you probably want to keep the default branch though?

Reply:
It was unnecessary according to kotlin compiler because there is only one type of action defined - MOVE_NORTH